### ISSUE TITLE
HW intrinsics shouldn't assert for unsupported

### ISSUE THREAD

**Issue by CarolEidt** [dotnet/coreclr#17815]
For example, here:
https://github.com/dotnet/coreclr/blob/master/src/jit/hwintrinsicxarch.cpp#L816
As described in the comments for https://github.com/dotnet/coreclr/pull/17737, this can lead to an assert in the case where the user knows the path will not be taken. Such cases should simply return `nullptr` so that the IL method will be invoked.

**Comment by fiigii** [comment]
Thanks for opening this issue. Perhaps, this assertion should be moved to codegen to avoid invalid instruction encoding.

**Comment by BruceForstall** [comment]
@CarolEidt Is this issue still valid, and needed for 3.0? Can you update this bug description to point to the assert(s) in question (or copy the relevant context of code into the bug description)? The code has changed since the issue was opened.

**Comment by CarolEidt** [comment]
This issue is no longer as relevant; the X86 intrinsics are fully exploded rather than being defined over T, and are defined as a class hierarchy (#19186), so I don't believe that it's possible to hit the assert, which is now here: https://github.com/dotnet/coreclr/blob/master/src/jit/hwintrinsicxarch.cpp#L849.
However, in the event that subsequent intrinsics are defined over `VectorX<T>` but don't support all possible base types, this path should return `nullptr` so that the IL method will be invoked. I propose to retain the assert so that we can ensure that such a case is flagged as a departure from the current approach.
@tannergooding @fiigii @eerhardt - do you know of a case where we have an intrinsic that's declared over `VectorX<T>` but not supported for all the possible base types?

**Comment by tannergooding** [comment]
> do you know of a case where we have an intrinsic that's declared over VectorX<T> but not supported for all the possible base types?

No, and based on the previous API review meeting; we will be exploding all future intrinsics in order to make it explicit what is and is not supported (such as when `Half` support comes online, or when we add other types in the future).

**Comment by fiigii** [comment]
> do you know of a case where we have an intrinsic that's declared over VectorX<T> but not supported for all the possible base types?

No. Currently, only platform-agnostic intrinsics could be generic, and they are not table-driven in importer/codegen.
